### DISCUSSION TITLE
Raspberry Pi - Slinger - Remote Login

### DISCUSSION THREAD

**Discussion by socelillyhammer** [GerryDazoo/Slinger#448] [Q&A]
Excuse my ignorance here and as a disclosure to some people I have checked other posts and have not found anything pertaining to this issue so I am creating this thread.

I need some help with Raspberry Pi - Remote Login.

Is there a way where I can remote login to my RPi and create a command that upon login to start the slinger server rather than have it running 24/7?

If so, can anyone provide guidance on how to do it?  I cant find anything and am not that tech savvy.

Thank you for the help in advance.

**Comment by robosmith** [comment]
If you have port forwarding enabled for WinSCP, you can connect with it or Putty remotely to edit files on your Pi to enable Slinger and reboot the Pi.
Same thing with shutting down. Edit the files to disable Slinger and reboot. Sorry I'm not sure which files need editing.

**Reply by socelillyhammer** [reply]
Thank you.  I've never heard of WinSCP or Putty.  But its a start and I will start reading up and learning.  Again, Thanks :-)

**Comment by GerryDazoo** [comment]
Firstly, what is the problem with it running 24/7?

You can SSH into the Pi  to run a command. be careful, because any process you start will die after the ssh session exits. Use nohup and put server into background. Google is your friend.

**Reply by socelillyhammer** [reply]
No problem at all with 24/7.  I thought maybe by letting it run 24/7 it would eat up RAM or memory loss or something.  Again, I am no where near as smart as you guys when it comes to coding, python, decryption, etc.  So my knowledge is highly limited - id say my level is a step above ignorance.  But if there is no issue with RAM, memory, etc. then for simplicity and ease I'm letting it run.

The only step thereafter is to create the VPN for seurity and I'll be happy.

My overall thinking was also along the lines that when I'm out of the country (I'm usually away a month or so at a time), if I need to reset or start up the slinger app I could have the ability to do so without having an issue from leaving it on 24/7

As for SSH command, nohup, again, I have to reserach and learn as this is all a foreign language, but I'm learning slowly.

Should be getting my RPi and starter kit today so the experminations and learning it all will be fun and tideous.

Again thank you for the guidance and tips.

**Reply by GerryDazoo** [reply]
No it's not going to eat up RAM. But if you want to restart slinger remotely just click on the "Restart" button on the remote page.

**Reply by socelillyhammer** [reply]
Awesome.  Thank you.  If I have any problems with the "Restart" button, I'll give a shout.

**Reply by socelillyhammer** [reply]
So for the remote access I  have 2 methods.

1. teamviewer - that would give you gui interface and you dont have to open ports
2. I did the SSH and read about commands and terminal on google - very very cool.  Thank you.  Seems that with SSH you need to open a port and on Pi you in the config you have to enable SSH as well for the port to be open.

Ive tested it out and ran the server via juicessh....wild stuff.  it worked.

Now I just need to open a port to stream remote and not use the SSH port and setup the VPN.

Very cool stuff.  Thank you

**Reply by mdehrlich** [reply]
I use xrdp started automatically on reboot by cron.  I use a windows laptop remote desktop to link.  I have no display, keyboard or mouse on my raspberry pi
https://raspberrytips.com/remote-desktop-raspberry-pi/
If you setup a port forward for the remote desktop, be careful you have a solid password.  Anyone hacking in has a lot of access to your network.
